executor/src/witgen/query_processor.rs

```diff
@@ -148,6 +148,17 @@ impl<'a, T: FieldElement> SymbolLookup<'a, T> for Symbols<'a, T> {
     }
 
     fn eval_challenge(&self, challenge: &Challenge) -> Result<Arc<Value<'a, T>>, EvalError> {
-        Ok(Value::FieldElement(self.fixed_data.challenges[&challenge.id]).into())
+        let challenge = *self
+            .fixed_data
+            .challenges
+            .get(&challenge.id)
+            .unwrap_or_else(|| {
+                panic!(
+                    "Challenge {} not found! Available challenges: {:?}",
```

Comment:
This can happen if a challenge is not used by any identity (and rightfully removed by the optimizer), but accessed by a hint.

Reply:
Can you create an issue to catch that in the analyzer?

Reply:
#1371